src/integer_mod_q/modulus/cmp.rs

```diff
@@ -47,6 +50,73 @@ impl PartialEq for Modulus {
 // This is not guaranteed by the [`PartialEq`] trait.
 impl Eq for Modulus {}
 
+impl PartialOrd for Modulus {
+    /// Compares two [`Modulus`] values. Used by the `<`, `<=`, `>`, and `>=` operators.
+    ///
+    /// Parameters:
+    /// - `other`: the other value that is used to compare the elements
+    ///
+    /// Returns the [`Ordering`] of the elements.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::Modulus;
+    ///
+    /// let a: Modulus = Modulus::from(10);
+    /// let b: Modulus = Modulus::from(42);
+    ///
+    /// assert!(a < b);
+    /// assert!(a <= b);
+    /// assert!(b > a);
+    /// assert!(b >= a);
+    /// ```
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Modulus {
+    //! Enables the usage of `max`, `min`, and `clamp`.
+    //!
+    //! # Examples
+    //! ```
+    //! use qfall_math::integer_mod_q::Modulus;
+    //! use std::cmp::{max, min};
+    //!
+    //! let a: Modulus = Modulus::from(10);
+    //! let b: Modulus = Modulus::from(42);
+    //!
+    //! assert_eq!(b, max(a.clone(), b.clone()));
+    //! assert_eq!(a, min(a.clone(), b.clone()));
+    //! assert_eq!(a, Modulus::from(2).clamp(a.clone(), b.clone()));
+    //! ```
+
+    /// Compares two [`Modulus`] values. Used by the `<`, `<=`, `>`, and `>=` operators.
+    ///
+    /// Parameters:
+    /// - `other`: the other value that is used to compare the elements
+    ///
+    /// Returns the [`Ordering`] of the elements.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::Modulus;
+    ///
+    /// let a: Modulus = Modulus::from(10);
+    /// let b: Modulus = Modulus::from(42);
+    ///
+    /// assert!(a < b);
+    /// assert!(a <= b);
+    /// assert!(b > a);
+    /// assert!(b >= a);
+    /// ```
+    fn cmp(&self, other: &Self) -> Ordering {
+        let z_1: Z = self.into();
+        let z_2: Z = other.into();
+        unsafe { fmpz_cmp(&z_1.value, &z_2.value).cmp(&0) }
+    }
```

Comment:
make to Z's like you have done and use the ordering implemented over Z. This way you list the access to the FFI